pkg/apis/numaflow/v1alpha1/vertex_types.go

```diff
@@ -233,10 +233,19 @@ func (v Vertex) GetPodSpec(req GetVertexPodSpecReq) (*corev1.PodSpec, error) {
 	}
 
 	spec := &corev1.PodSpec{
-		Subdomain:      v.GetHeadlessServiceName(),
-		Volumes:        append(volumes, v.Spec.Volumes...),
-		InitContainers: v.getInitContainers(req),
-		Containers:     append(containers, v.Spec.Sidecars...),
+		NodeSelector:       v.Spec.NodeSelector,
```

Comment:
`ApplyToPodSpec` was filling this earlier?

Reply:
> ApplyToPodSpec

oops, you are right!